include/nbl/builtin/hlsl/tgmath.hlsl

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_TGMATH_INCLUDED_
+#define _NBL_BUILTIN_HLSL_TGMATH_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/ieee754.hlsl>
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+
+#ifdef __HLSL_VERSION
+#include <nbl/builtin/hlsl/spirv_intrinsics/core.hlsl>
+#endif
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace tgmath
```

Comment:
there's no `tgmath` namespace in `std`

Reply:
also in `std::` the isnan and isinf don't come from tgmath AFAIK, why did we put them in tgmath and not cpp_comapat?

Reply:
moved specializations of `isinf` and `isnan` from `tgmath.hlsl` to `intrinsics.hlsl`. Now i don't remember what we added `tgmath.hlsl` for xd